google/ggrc-core
# GGRC-5985 Allow external users work on behalf their account

# Issue description

We need to be able to provide GGRCQ service account rights to any GGRC user on call from GGRCQ as a trusted source.

The behavior:
1. New x-external-user header should be added. 
2. 
  - If the user doesn't exist in GGRC, create new user with Global Creator role.
  - If exists use the existing user
3. Create the object requested by the external system. Permissions should be checked by service account
4. Put in modified by field the user that was specified in x-external-user header.
5. Corresponding revisions needs to be created

**Note**: Please keep backward compatibility. Request can be done without X-external-user header, in that case X-external-user will equals to X-ggrc-user.

# Steps to test the changes

Use the postman collection for testing that you can find in JIRA.
Set ALLOWED_QUERYAPI_APP_IDS and EXTERNAL_APP_USER according to the test data in the postman collection. 

At the moment GGRCQ already creating Relationships and Person object types via the service account. Please check that it will work.

# Solution description

Login on behalf of new header if it is exists

# Sanity checklist




# PR Review checklist
